Feature/dcifer

a-hubbard opened the pull request:
## Pull Request for PGEcore

Thank you for your contribution to PGEcore!✨

### Description of changes

This is the wrapper script for Dcifer. It should be functional and have all necessary documentation. It has not been tested with providing external allele frequencies, as we currently don't have an example file for that.

### Pull Request Checklist

- [x] I have provided a description and detailed information about the changes made
- [x] I have followed the [PGEcore Code Guidelines](https://github.com/PlasmoGenEpi/PGEcore?tab=readme-ov-file#code-guidelines)
- [x] I have followed the [PGEcore how to contribute instructions](https://github.com/PlasmoGenEpi/PGEcore?tab=readme-ov-file#how-to-contribute) and I am merging into the `develop` branch
- [x] Documentation is updated, if necessary
- [x] Relevant issues are linked, if applicable

m-murphy commented:
Can you change the way you detect cores to using `parallelly::availableCores`? see here for why `detectCores` is bad news https://www.jottr.org/2022/12/05/avoid-detectcores/

a-hubbard commented:
I switched specimen_a (and b) to specimen_id_a, updated all input and output path arguments to make sure they say _table or something like that, and replaced `detectCores` with `parallelly::availableCores()`.

kathryn1995 commented:
I am struggling to run this now which we could talk about in person tomorrow. I am thinking it is due to the switch in detecting cores. I think it may be better to specify the number of cores on the command line as this would be better for running in Nextflow. Could you also change btwn_host_rel_table to btwn_host_rel_output or output_path to make it clear this is where to output and isn't a table that needs to be input

a-hubbard commented:
I updated the single locus allele frequency example for microhaplotype data and tested the user-provided allele frequency functionality in the wrapper. It is now ready for review.
